### PR TITLE
Integration test workflow

### PR DESCRIPTION
**Pull request by majguo**
## Summary

The PR includes the following major changes:

* Add one workflow **Integration test** `integ-test.yaml` by leveraging `docker-compose`
* Run additional integration test for the existing workflow **Setup demo**
* Run additional integration test for the existing workflow **Rolling update**
* Setup Elastic/Kibana/Filebeat containers in `docker-compose`

## Testing

New/updated workflows ran successfully in my fork:

* [Integration test](https://github.com/majguo/piggymetrics/actions/runs/6859427692)
* [Setup demo](https://github.com/majguo/piggymetrics/actions/runs/6858525476): integration test is bypassed as gateway is not accessible if the app is deployed using internal sub
* [Setup demo](https://github.com/majguo/piggymetrics/actions/runs/6858153729): integration test is executed
* [Rolling update](https://github.com/majguo/piggymetrics/actions/runs/6858631593): integration test is bypassed as gateway is not accessible if the app is deployed using internal sub
* [Rolling update](https://github.com/majguo/piggymetrics/actions/runs/6858413875): integration test is executed

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -264,7 +267,7 @@ docker login ghcr.io -u <your-github-username> -p <your-github-personal-access-t
 
 export KEYSTORE_PASSWORD=<your-keystore-password>
 export IMAGE_TAG=1.0
-./build.sh $KEYSTORE_PASSWORD $REGISTRY_NAME $IMAGE_TAG ${NGINX_VERSION} ${OTEL_MODULE_VERSION}
+./build.sh $KEYSTORE_PASSWORD $IMAGE_TAG ${NGINX_VERSION} ${OTEL_MODULE_VERSION} $REGISTRY_NAME
```
Comment (edburns) [inline README.md]: I find `build.sh` is too small a name for what it does. I suggest renaming `build.sh`  to something more accurate, such as `build-piggymetrics-microservices.sh`. What do you think?

File: build.sh [hunk 2]
```diff
@@ -1,10 +1,10 @@
 #!/bin/bash
```
Comment (edburns) [inline build.sh]: See my comment on `README.md` about "I find `build.sh` is too small a name".

**Review by edburns — CHANGES_REQUESTED** [review]
See comment on `README.md` and `build.sh`.

File: README.md (hunk 1 above)
Comment (majguo) [reply README.md]: Good suggestion, but I prefer a shorter name, such as `build-piggymetrics.sh`.

File: build.sh (hunk 2 above)
Comment (majguo) [reply build.sh]: Addressed in [rename to build-piggymetrics.sh](https://github.com/Azure-Samples/piggymetrics/pull/13/commits/4d324c6b6f074b2600bcafc67d5495c417ecc22e).

**Review by edburns — APPROVED** [review]

**Pull request merged**